include/zephyr/net/ieee802154_radio_ext_ot.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *   This file extends interface of ieee802154_radio.h for OpenThread.
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_IEEE802154_RADIO_EXT_OT_H_
+#define ZEPHYR_INCLUDE_NET_IEEE802154_RADIO_EXT_OT_H_
+
+#include <zephyr/net/ieee802154_radio.h>
+
+/**
+ *  OpenThread specific capabilities of ieee802154 driver.
+ *  This type extends ieee802154_hw_caps.
+ */
+enum ieee802154_ext_ot_hw_caps {
+	IEEE802154_EXT_OT_HW_MULTIPLE_CCA = BIT(IEEE802154_HW_CAPS_BITS_PRIV_START),
+};
+
+enum ieee802154_ext_ot_tx_mode {
+	/**
+	 * Transmit packet in the future, perform one or more CCA before transmission.
+	 * The number of CCA attempts to be performed is at most 1 + extra_cca_attempts.
+	 * The CCA attempts are performed back-to-back until CCA returns idle channel or
+	 * until 1 + extra_cca_attempts CCA attempts are performed.
+	 * The extra_cca_attempts is set by IEEE802154_EXT_OT_CONFIG_MULTIPLE_CCA configuration.
+	 * Requires IEEE802154_EXT_OT_HW_MULTIPLE_CCA capability.
+	 */
+	IEEE802154_EXT_OT_TX_MODE_TXTIME_MULTIPLE_CCA = IEEE802154_TX_MODE_PRIV_START
+};
+
+/**
+ *  OpenThread specific configuration types of ieee802154 driver.
+ *  This type extends ieee802154_config_type.
+ */
+enum ieee802154_ext_ot_config_type {
+	/** Allows to configure extra CCA attempts for transmission requested with mode
+	 *  IEEE802154_EXT_OT_TX_MODE_TXTIME_MULTIPLE_CCA.
+	 *  Requires IEEE802154_EXT_OT_HW_MULTIPLE_CCA capability.
+	 */
+	IEEE802154_EXT_OT_CONFIG_MULTIPLE_CCA = IEEE802154_CONFIG_PRIV_START
+};
+
+/** OpenThread specific configuration data of ieee802154 driver. */
+struct ieee802154_ext_ot_config {
+	union {
+		struct ieee802154_config base;
+
+		/** ``IEEE802154_EXT_OT_CONFIG_MULTIPLE_CCA``
+		 *
+		 *  The maximum number of extra CCA attempts to be performed when transmission is
+		 *  requested with mode IEEE802154_EXT_OT_TX_MODE_TXTIME_MULTIPLE_CCA.
```

Comment:
Clarify between two options? I assume it's first one, but worth making it clear.

1. when transmission is requested to the radio driver by the upper layer
2. when transmission is requested to the lower layer by the radio driver (after the desired delay)

Reply:
I agree that the timing requirements should be documented (as we agreed this shows the way and is a precondition for a potential future soft MAC solution w/o the need to implement it right away). The easiest way would be to just point to the (future) standards clause but that does not seem to be possible, so alternatively the actual requirements need to be documented IMO:

1) For the other timed TX modes, the TX timestamp points to the expected start of the first symbol of the PHY header (i.e. after CCA completed) for packets scheduled to a future point in time. What does the TX timestamp point to here? The first symbol of the PHY header assuming the first CCA attempt shows a free medium? The second CCA attempt if the first was showing a busy medium? The first CCA attempt?

2) It has been variously stated that a cross-vendor soft MAC approach was not possible due to strict timing requirements for the extra CCA attempts. What are these strict requirements? What is the duration of each CCA attempt? How are the CCA attempts scheduled? Or alternatively: What are the requirements re guard times between the attempts? Which tolerances (drift, jitter) are acceptable?

3) How is CCA performed (ie. which of the standard CCA modes are allowed here)?

Docs tend to become outdated and redundant docs even more quickly: I therefore propose that we keep the explanation as precise as possible but only in the TX_MODES enum and @ref to it here.

Reply:
I agree, that needs to be addressed. I will return with proposal.